fe/fe-core/src/main/java/org/apache/doris/nereids/rules/rewrite/PushDownAggThroughJoinOneSide.java

```diff
@@ -80,7 +80,8 @@ public List<Rule> buildRules() {
                         .thenApply(ctx -> {
                             Set<Integer> enableNereidsRules = ctx.cascadesContext.getConnectContext()
                                     .getSessionVariable().getEnableNereidsRules();
-                            if (!enableNereidsRules.contains(RuleType.PUSH_DOWN_AGG_THROUGH_JOIN_ONE_SIDE.type())) {
+                            if (!enableNereidsRules.contains(RuleType.PUSH_DOWN_AGG_THROUGH_JOIN_ONE_SIDE.type())
+                                    && !RuleType.PUSH_DOWN_AGG_THROUGH_JOIN_ONE_SIDE.checkUseCboRuleHint()) {
```

Comment:
the mod of cost based rewrite is not enough when we want just two states: (hint use it , close it), because in cost based rewrite we have three states (hint use it , cost based rewrite, hint close it). So this means we change (cost based rewrite, hint close it) states to close, then this would not change behavior controlled by  session variables